utils/vlog/vlog_to_pbtype.py

```diff
@@ -54,68 +54,225 @@
 
 sys.path.insert(0, "..")
 from lib import xmlinc
+from sdf_timing import sdfparse
 
-parser = argparse.ArgumentParser(
-    description=__doc__.strip(), formatter_class=argparse.RawTextHelpFormatter
-)
-parser.add_argument(
-    'infiles',
-    metavar='input.v',
-    type=str,
-    nargs='+',
-    help="""\
-One or more Verilog input files, that will be passed to Yosys internally.
-They should be enough to generate a flattened representation of the model,
-so that paths through the model can be determined.
-"""
-)
-parser.add_argument(
-    '--top',
-    help="""\
-Top level module, will usually be automatically determined from the file name
-%.sim.v
-"""
-)
-parser.add_argument(
-    '--includes',
-    help="""\
-Command seperate list of include directories.
-""",
-    default=""
-)
-parser.add_argument('-o', help="""\
-Output filename, default 'model.xml'
-""")
 
-args = parser.parse_args()
-iname = os.path.basename(args.infiles[0])
+def make_timings(
+        pb_type_xml, sdf_file_name, sdf_cell_paths=None, sdf_variant="slow"
+):
+    """
+    Loads timings from a given SDF file and converts them to appropriate
+    statements within pb_type XML.
 
-outfile = "pb_type.xml"
-if "o" in args and args.o is not None:
-    outfile = args.o
+    If sdf_cell_name is None then the function tries to find in SDF cell with
+    name equal to the pb_type name.
 
-yosys.run.add_define("PB_TYPE")
-if args.includes:
-    for include in args.includes.split(','):
-        yosys.run.add_include(include)
+    If sdf_inst_name is None and there is only one instance of that cell in SDF
+    then that one is taken.
+    """
 
-vjson = yosys.run.vlog_to_json(args.infiles, flatten=False, aig=False)
-yj = YosysJSON(vjson)
+    def parse_timescale(timescale_str):
```

Comment:
this function is a part of python-sdf-timing lib. Why do you duplicate the code here instead of using it directly from the lib?

Reply:
Done.